src/commands/posts-list/open-post-in-vscode.ts

```diff
@@ -8,13 +8,22 @@ import { PostFileMapManager } from '../../services/post-file-map';
 import { Settings } from '../../services/settings.service';
 import { openPostFile } from './open-post-file';
 import { PostTitleSanitizer } from '../../services/post-title-sanitizer.service';
+import { postCategoryService } from '../../services/post-category.service';
 
 const buildLocalPostFileUri = async (post: Post, includePostId = false): Promise<Uri> => {
     const workspaceUri = Settings.workspaceUri;
+    const saveWithC = Settings.saveWithC;
     const ext = `.${post.isMarkdown ? 'md' : 'html'}`;
     const postIdSegment = includePostId ? `.${post.id}` : '';
     const { text: postTitle } = await PostTitleSanitizer.sanitize(post);
-    return Uri.joinPath(workspaceUri, `${postTitle}${postIdSegment}${ext}`);
+    if (saveWithC) {
+        let categories = await postCategoryService.fetchCategories();
+        categories = categories.filter(x => post.categoryIds?.includes(x.categoryId));
+        const categoryTitle = categories.length > 0 ? categories.map(c => c.title).join(',') : '未分类';
```

Comment:
May be it's better to save post with no category to the the root blogs workspace folder